database/src/main/java/com/firebase/ui/database/CachingObservableSnapshotArray.java

```diff
@@ -43,11 +43,19 @@ public T getObject(int index) {
         }
     }
 
-    protected void clearData() {
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
         getSnapshots().clear();
         mObjectCache.clear();
+
+        clearData();
     }
 
+    /** @deprecated use {@link ObservableSnapshotArray#onDestroy()} instead */
+    @Deprecated
+    protected void clearData() {}
```

Comment:
In order to not make this a breaking change this method should still actually clear the data.